src/RooBernsteinSideband.cxx

```diff
@@ -0,0 +1,625 @@
+           ret   +=((RooAbsReal&) _coefList[ipar]).getVal()*bernknintgbinx*bernknintgbiny*bernknintgbinz;
+           intg_1+=((RooAbsReal&) _coefList[ipar]).getVal();
+//
+	   ipar++;
+	  }
+         }
+       }
+     //okkio!!! if(ret<1.E-30) ret = 1.E-30;
+    intg_1 = (1.0+maxDegree1)*(1.0+maxDegree2)*(1.0+maxDegree3)/intg_1;
+//    ret=ret/(xBinw*yBinw*zBinw);
+//P.S. Il termmine xdif*ydif*zdif si elide sopra e sotto
+      ret = ret*intg_1;
+//    std::cout<<"ret = "<<ret<<std::endl;
+   return ret;
+
+ }
+fptype  RooBernsteinSideband::device_SidebandBernsteinkn_intgBin( fptype xLeft, fptype xRight, fptype enne, fptype kappa) const{
+
+      if(xLeft==0.0&&xRight==1.0) return 1.0/(enne+1.0);
+ 
+      fptype integbernkn = 0.0;
+      fptype ifactni = 0.0;
+      fptype ifactik = 0.0;
+      
+      
+      fptype powxL = pow(xLeft ,kappa+1)   ;
+      fptype powxR = pow(xRight,kappa+1) ;
+       for(fptype i = kappa; i <=enne ; ++i) {
+// n!/(i!(n-i)!)
+
+//        ifactni =  device_coeffbinomial(enne,i);
+        ifactni =  TMath::Binomial(enne,i);
+// i!/(k!(i-k)!)
+ 
+        ifactik =  TMath::Binomial(i,kappa);
+//        ifactik =  device_coeffbinomial(i,kappa);
+        integbernkn += ifactni*ifactik*pow(-1.0,i-kappa)*(powxR-powxL)/(i+1);
+	powxL*=xLeft ;
+	powxR*=xRight;
+//        integbernkn += ifactni*ifactik*pow(-1.0,i-kappa)*(pow(xRight, i+1)-pow(xLeft,i+1))/(i+1);
+       }
+
+       if (integbernkn<=0.0 ){
+          printf(" Error in SidebandBernsteinkn_intgbin xLeft=%f xRight=%f kappa=%f enne=%f integral = %5.15f\n",xLeft,xRight,kappa,enne,integbernkn);
+        integbernkn=1.E-30;
+       }
+       return integbernkn;
+}
+ 
+Int_t RooBernsteinSideband::getAnalyticalIntegral(RooArgSet& allVars, RooArgSet& analVars, const char* /*rangeName*/) const
+{
+  if       (matchArgs(allVars, analVars, _x,_y,_z)){
+//    std::cout<<"getAnalyticalIntegral==1: Analytic integral over x y z"<<std::endl;
+   return 1;
+  }else if (matchArgs(allVars, analVars, _y,_z)){
+//    std::cout<<"getAnalyticalIntegral==2: Analytic integral over y z"<<std::endl;
+   return 2;
+  }else if (matchArgs(allVars, analVars, _x,_z)){
+//    std::cout<<"getAnalyticalIntegral==3: Analytic integral over x z"<<std::endl;
+   return 3;
+  }else if (matchArgs(allVars, analVars, _x,_y)){
+//    std::cout<<"getAnalyticalIntegral==4: Analytic integral over x y"<<std::endl;
+   return 4;
+  }else if (matchArgs(allVars, analVars, _x   )){
+//    std::cout<<"getAnalyticalIntegral==5: Analytic integral over x  "<<std::endl;
+   return 5;
+  }else if (matchArgs(allVars, analVars, _y   )){
+//    std::cout<<"getAnalyticalIntegral==6: Analytic integral over y  "<<std::endl;
+   return 6;
+  }else if (matchArgs(allVars, analVars, _z   )){
+//    std::cout<<"getAnalyticalIntegral==7: Analytic integral over z  "<<std::endl;
+   return 7;
+  }else{  
+   std::cout << "Error in RooBernsteinSideband::analyticalIntegral: set getAnalyticalIntegral==0" << std::endl;
```

Comment:
```suggestion
//   std::cout << "Error in RooBernsteinSideband::analyticalIntegral: set getAnalyticalIntegral==0" << std::endl;
```
Apparently, the integral is sometime called during a fit without any integration variable. I we don't believe this is a problem I would remove the error message

Reply:
I noticed this and I want to remove it too.